src/Services/Container.php

```diff
@@ -39,10 +53,15 @@ public static function get() {
 
   /**
    * Create a new loader instance.
+   *
+   * @param array $specs
+   *   The initial set of specs.
+   * @param string $name
+   *   The service name of the container itself. (default: 'container').
    */
-  public function __construct($specs = [], $name = 'container') {
+  public function __construct(array $specs = [], string $name = 'container') {
     $this->instances[$name] = $this;
-    $this->specs = $specs;
+    $this->specs = $this->processSpecs($specs);
```

Comment:
Since defaults can only be set in `setDefaults()` they can never be applied to `$specs` passed to the constructor.

The constructor argument is only used by the tests at the moment. It could be removed in favor of `addSpecs()`.